andela/nightcrawler-backend
# #167727310 Endpoints for Social Media Login

#### What does this PR do?
This PR build the endpoints for social media login. A user can log in to barefoot nomad via facebook or google
#### Description of Task to be completed?

- The following endpoint was set up to handle google and facebook login respectively
 GET api/v1/auth/google
 GET api/vi/auth/facebook
- Setup google login app on Google developer console

- Setup facebook login app on Facebook developer console 

- Setup passport to use both google and facebook strategy

#### How should this be manually tested?

- Run git clone in terminal `https://github.com/andela/nightcrawler-backend.git`
- Cd into the folder created `nightcrawler-backend` and run git fetch `ft--social-media-login-167727310`, finally run npm install
- Make sure your database is seeded with a user that has a valid google/facebook account
- In your .env file add `GOOGLE_CLIENT_ID`, `GOOGLE_CLIENT_SECRET` ,`GOOGLE_CALLBACK` , `FACEBOOK_APP_ID`,  `FACEBOOK_APP_SECRET`, `FACEBOOK_CALLBACK` as shared in the slack channel
- Run npm run start:dev to start the server
- Navigate to `localhost:300/api/v1/auth/google` for google login or `localhost:300/api/v1/auth/facebook`
- Login into google or facebook with an email address registered on barefoot nomad
#### Any background context you want to provide?
- Your database and model needs to be in place.
- enviroment variable as stated above also needs to available
#### What are the relevant pivotal tracker stories?
#167727310
